extralit/src/extralit/cli/documents/embed.py

```diff
@@ -0,0 +1,355 @@
+            current_chunk = current_chunk[max(0, break_point - overlap) :]
+
+    # Add final chunk if there's remaining content
+    if current_chunk.strip():
+        chunks.append(
+            {
+                "id": str(uuid4()),
+                "content": current_chunk.strip(),
+                "metadata": {
+                    "chunk_index": chunk_index,
+                    "page_number": current_page,
+                    "header": current_headers[-1]["text"] if current_headers else "",
+                    "level": current_headers[-1]["level"] if current_headers else 0,
+                    "header_hierarchy": [h["text"] for h in current_headers],
+                },
+            }
+        )
+
+    return chunks
+
+
+def create_embedding(text: str, model: Optional[str] = None) -> Optional[list[float]]:
+    """
+    Create embedding for text using configurable endpoint and model.
+
+    Args:
+        text: Text to embed
+        model: Embedding model to use (overrides EMBED_MODEL env var)
+
+    Returns:
+        List of float values representing the embedding, or None if failed
+    """
+    if not OPENAI_API_KEY or OPENAI_BASE_URL == "random" or not OPENAI_BASE_URL.startswith("http"):
+        # Generate random 1536-dimensional vector (same as text-embedding-ada-002)
+        embedding = np.random.rand(1536).tolist()
+        return embedding
+
+    from llama_index.embeddings.openai import OpenAIEmbedding
+
+    embed_model = OpenAIEmbedding(model=model or EMBED_MODEL_NAME, api_key=OPENAI_API_KEY, base_url=OPENAI_BASE_URL)
+
+    # Get embedding
+    embedding = embed_model.get_text_embedding(text)
+    return embedding
+
+
+def create_records_from_chunks(document, chunks: list[dict[str, Any]]) -> list[dict[str, Any]]:
+    """
+    Create dataset records from document chunks.
+
+    Args:
+        document: Document object with metadata
+        chunks: List of chunk dictionaries
+
+    Returns:
+        List of record dictionaries ready for dataset logging
+    """
+    records = []
+
+    for chunk in chunks:
+        # Create embedding for chunk content
+        embedding = create_embedding(chunk["content"])
+        if embedding is None:
+            continue
+
+        # Prepare record
+        record = {
+            "fields": {
+                "header": chunk["metadata"]["header"],
+                "content": chunk["content"],
+            },
+            "metadata": {
+                "reference": document.reference or str(document.id),
+                "doc_id": str(document.id),
+                "chunk_index": chunk["metadata"]["chunk_index"],
+                "page_number": chunk["metadata"]["page_number"],
+                "header": chunk["metadata"]["header"],
+                "level": chunk["metadata"]["level"],
+                "header_hierarchy": " > ".join(chunk["metadata"]["header_hierarchy"]),
+            },
+            "vectors": {"content": embedding},
+        }

```

Comment:
Let's keep the record structure this way. Metadata is used for record filter and sorting, whereas the `fields` are for displaying content.